lib/modules/platform/bitbucket/pr-cache.ts

```diff
@@ -0,0 +1,121 @@
+import { dequal } from 'dequal';
+import { DateTime } from 'luxon';
+import { logger } from '../../../logger';
+import * as memCache from '../../../util/cache/memory';
+import { getCache } from '../../../util/cache/repository';
+import type { BitbucketHttp } from '../../../util/http/bitbucket';
+import type { Pr } from '../types';
+import type { BitbucketPrCacheData, PagedResult, PrResponse } from './types';
+import { prFieldsFilter, prInfo, prStates } from './utils';
+
+export class BitbucketPrCache {
+  private cache: BitbucketPrCacheData;
+
+  private constructor(
+    private repo: string,
+    private author: string | null,
+    private prList: Pr[] = [],
+  ) {
+    const repoCache = getCache();
+    repoCache.platform ??= {};
+    repoCache.platform.bitbucket ??= {};
+
+    let pullRequestCache: BitbucketPrCacheData | undefined =
+      repoCache.platform.bitbucket.pullRequestsCache;
+    if (!pullRequestCache || pullRequestCache.author !== author) {
+      pullRequestCache = {
+        items: {},
+        updated_on: null,
+        author,
+      };
+    }
+    repoCache.platform.bitbucket.pullRequestsCache = pullRequestCache;
+    this.cache = pullRequestCache;
+  }
+
+  static async init(
+    http: BitbucketHttp,
+    repo: string,
+    author: string | null,
+  ): Promise<BitbucketPrCache> {
+    const prList = memCache.get<Pr[] | undefined>('bitbucket-pr-cache');
```

Comment:
why checking memcache here, I don't see where it's written. I don't think we need that additional cache layer.